lib/backport_session.js

```diff
@@ -226,4 +247,51 @@ export default class BackportSession extends Session {
       };
     });
   }
+
+  getCurrentBranch() {
+    return runSync('git',
+      ['rev-parse', '--abbrev-ref', 'HEAD']
+    ).trim();
+  }
+
+  updateUpstreamRefs(branchName) {
+    runSync('git',
+      ['fetch', 'upstream', branchName]
+    );
+  }
+
+  getBranchCommit(branch) {
+    return runSync('git',
+      ['rev-parse', branch]
+    ).trim();
+  }
+
+  isBranchUpToDateWithUpstream(branch) {
+    this.updateUpstreamRefs(branch);
+
+    const localCommit = this.getBranchCommit(branch);
+    const upstreamCommit = this.getBranchCommit(`upstream/${branch}`);
+
+    return localCommit === upstreamCommit;
+  };
+
+  isLocalBranchExists(branch) {
+    try {
+      // will exit with code 1 if branch does not exist
+      runSync('git',
+        ['rev-parse', '--verify', '--quiet', branch]
+      );
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  syncBranchWithUpstream(branch) {
+    const currentBranch = this.getCurrentBranch();
+
+    runSync('git',
+      [currentBranch !== branch ? 'fetch' : 'pull', 'upstream', `${branch}:${branch}`, '-f']
+    );
```

Comment:
```suggestion
    runSync('git', ['fetch', upstream, branch]);
    if (currentBranch === branch) runSync('git', ['reset', 'FETCH_HEAD', '--hard']);
```
I'm not even sure the `if` condition is necessary, but I guess I'm not using git like everyone else 🤷‍♂️

Reply:
the fetch here will not update the local branch...

Reply:
I don't use local branches, so I guess that's why this looks weird to me.